packages/react/src/use-table/types.ts

```diff
@@ -29,20 +29,26 @@ export type TableColumn = {
   /** Identifier for the column */
   identifier: string;
   /** Human-readable header value for the column */
-  header: string;
-  /** Dot-separated path to the field in the record */
-  field: string;
-  type: ColumnType;
+  header: ReactNode;
   /** parent relationship type if the parent field is a relationship */
   relationshipType?: RelationshipType;
   sortable: boolean;
   /** For controlling if the time is shown on DateTime cell renderers   */
   includeTime?: boolean;
-  /** Custom render function */
-  render?: CustomCellRenderer;
```

Comment:
Just curious, whahappensen right now if I:
A. The `type` is `GadgetFieldType` and the `render` property exists, or...
B. The `type` is `CustomRenderer` and the `field` property exists